pkg/cmd/grafana-cli/commands/commands.go

```diff
@@ -10,30 +10,34 @@ import (
 	"github.com/grafana/grafana/pkg/cmd/grafana-cli/commands/datamigrations"
 	"github.com/grafana/grafana/pkg/cmd/grafana-cli/commands/secretsmigrations"
 	"github.com/grafana/grafana/pkg/cmd/grafana-cli/logger"
-	"github.com/grafana/grafana/pkg/cmd/grafana-cli/runner"
 	"github.com/grafana/grafana/pkg/cmd/grafana-cli/services"
 	"github.com/grafana/grafana/pkg/cmd/grafana-cli/utils"
 	"github.com/grafana/grafana/pkg/infra/db"
 	"github.com/grafana/grafana/pkg/infra/tracing"
+	"github.com/grafana/grafana/pkg/server"
 	"github.com/grafana/grafana/pkg/services/sqlstore/migrations"
 	"github.com/grafana/grafana/pkg/setting"
 )
 
-func runRunnerCommand(command func(commandLine utils.CommandLine, runner runner.Runner) error) func(context *cli.Context) error {
+func runRunnerCommand(command func(commandLine utils.CommandLine, runner server.Runner) error) func(context *cli.Context) error {
 	return func(context *cli.Context) error {
 		cmd := &utils.ContextCommandLine{Context: context}
-
-		cfg, err := initCfg(cmd)
+		configOptions := strings.Split(cmd.String("configOverrides"), " ")
+		runner, err := server.InitializeForCLI(setting.CommandLineArgs{
+			Config:   cmd.ConfigFile(),
+			HomePath: cmd.HomePath(),
+			// tailing arguments have precedence over the options string
+			Args: append(configOptions, cmd.Args().Slice()...),
+		})
 		if err != nil {
-			return fmt.Errorf("%v: %w", "failed to load configuration", err)
+			return fmt.Errorf("%v: %w", "failed to initialize runner", err)
 		}
 
-		r, err := runner.Initialize(cfg)
-		if err != nil {
-			return fmt.Errorf("%v: %w", "failed to initialize runner", err)
+		if cmd.Bool("debug") {
+			runner.Cfg.LogConfigSources()
```

Comment:
Any particular reason to inline this in 2 places?

Reply:
Previously it's been a shared function, now since the order of injection has changed (instead of Cfg we take CommandLineArgs, and use Cfg from the instantiated Runner) - I decided to have this code duplicated. But yeah, there is a way to unify this again.